Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -641,6 +655,179 @@ namespace gdjs {
+          // Revert side effect on the search context.
+          context.revertTo(previousAllowedMinDeltaY, previousAllowedMaxDeltaY);
+          continue;
+        }
+        if (context.isCollidingAnyPlatform()) {
+          isCollidingAnyPlatform = true;
+        }
+        if (context.floorIsTooHigh()) {
+          // One platform is colliding the character
+          // and is too high for the character to walk on.
+          // This will still be an obstacle even if there
+          // are other platforms that fit the requirements.
+          highestGround = null;
+          break;
+        }
+
+        if (
+          context.isCollidingAnyPlatform() &&
+          highestRelativeY < totalHighestY
+        ) {
+          totalHighestY = highestRelativeY;
+          highestGround = platform;
+        }
+      }
+      if (highestGround) {
+        const object = this.owner;
+        object.setY(object.getY() + totalHighestY);
+      }
+      const returnValue =
+        gdjs.PlatformerObjectRuntimeBehavior._platformSearchResult;
+      returnValue.highestGround = highestGround;
+      returnValue.isCollidingAnyPlatform = isCollidingAnyPlatform;
+      return returnValue;
+    }
+
+    /**
+     * Find the highest Y relative to the owner bottom of the floor reachable by the owner.
+     * @param platform The platform to be tested for collision.
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     * @return the search context
+     */
+    private _findPlatformHighestRelativeYUnderObject(
+      platform: gdjs.PlatformRuntimeBehavior,
+      context: FollowConstraintContext
+    ): FollowConstraintContext {
+      const platformObject = platform.owner;
+      const platformAABB = platformObject.getAABB();
+      if (
+        platformAABB.max[0] <= context.ownerMinX ||
+        platformAABB.min[0] >= context.ownerMaxX ||
+        platformAABB.max[1] <= context.headMinY ||
+        platformAABB.min[1] > context.floorMaxY
+      ) {
+        // No collision
+        return context;
+      }
+
+      for (const hitbox of platformObject.getHitBoxes()) {
+        let previousVertex = hitbox.vertices[hitbox.vertices.length - 1];
+
+        // Edges over the character head might not result to a collision,
+        // but if there is also an edge under its head then there is a collision.
+        // The platform hitbox could be in several parts.
+        // So, the object could walk on one part
+        // and have another part over its head.
+        context.initializeBeforeHitboxCheck();
+
+        for (const vertex of hitbox.vertices) {
+          const deltaX = vertex[0] - previousVertex[0];
+          if (
+            // The vertex is into the interval excluding bounds
+            (context.ownerMinX < vertex[0] && vertex[0] < context.ownerMaxX) ||
+            // or the on a bound but its edge is from the inside.
```

Comment:
typo here

Reply:
That's interesting, why are we checking the precise equality of this edge on a bound? I.e: could you explain why we're excluding bounds in the first condition? :)